### PR TITLE
Add event log fetching and parsing

### PR DESCRIPTION
**Pull request by alexmwu**
Here we add support for grabbing the Linux TCG event log and use the work https://github.com/google/go-attestation does to parse the event log.
This includes a unit test against a cached event log and  a system test, supporting a real event log and TPM.

Tested:
`sudo ../server.test  -test.run TestSystemParseEventLog -tpm-path /dev/tpmrm0 -event-log-path /sys/kernel/security/tpm0/binary_bios_measurements`
on a physical machine and a Shielded VM.

### CODE REVIEW

**Comment by alexmwu** [comment]
Rebased on the tpm2tools->client package name change

File: server/test/debian_10_event_log.yml [hunk 1]
```diff
@@ -0,0 +1,164 @@
+// Fetched via `tpm2_eventlog debian_10_binary_bios_measurements`
```
Comment (josephlr) [inline server/test/debian_10_event_log.yml:1]: Let's just remove this for now, as we do not use it in our tests.

File: server/test/debian_10_event_log.yml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+// Note this is invalid yaml.
+---
+events:
```
Comment (josephlr) [inline server/test/debian_10_event_log.yml:4]: We should use output from a Newer GCP VM to at least get the EV_NONHOST_INFO stuff.

Comment (alexmwu) [reply server/test/debian_10_event_log.yml]: Done.

File: server/test/debian_10_event_log.yml (hunk 1 above)
Comment (alexmwu) [reply server/test/debian_10_event_log.yml]: Done.

**Comment by josephlr** [comment]
@alexmwu 

Rebased on latest master, and slightly change the dependancy importing commit, will review the rest tomorrow.

It also seems like there's a test import cycle, can you fix that?

**Comment by alexmwu** [comment]
> @alexmwu
> 
> Rebased on latest master, and slightly change the dependancy importing commit, will review the rest tomorrow.
> 
> It also seems like there's a test import cycle, can you fix that?

weird, I thought I pushed that change. fixed it now

**Comment by josephlr** [comment]
@googlebot I consent.

**Comment by josephlr** [comment]
@alexmwu I was able to simplify the circular dependency issue by moving the client tests to a `client_test` crate (while keeping everything in the same place) see https://github.com/google/go-tpm-tools/pull/100/commits/9a6cfa4fe6a11ba073cb71558d77b3f3badce770.

This required some other changes (see the commits before that one):
  - Making the `tpmpb.Pcrs` functions methods instead
  - Making some values public

I also shortened and cleaned up some of the eventlog_test code (mostly just moving stuff inline)

Also, do you have the corresponding PCRs for the RHEL tests?

**Review by josephlr — APPROVED** [review]

**Pull request merged**